BJS2-55521

lexsandr666 opened the pull request:
Added env and changed the configuration of application.yaml

lexsandr666 committed:
BJS2-55521 : added env and changed the configuration of application.yaml

lexsandr666 committed:
BJS2-55521 : added env and changed the configuration of application.yaml

ZhannaD commented on src/main/java/school/faang/user_service/config/EnvHelper.java:
```diff
@@ -0,0 +1,17 @@
+    public static void loadVariables() {
+        Dotenv dotenv = Dotenv.configure()
+            .directory("src/main/resources")
+            .ignoreIfMissing()
```
I think the .env file should be placed in the root directory as an application file.

lexsandr666 replied:
Если бы все конфигурационные файлы лежали в корене приложения, то возможно это было бы правильно. Но так как у нас yaml файл лежил в файлах ресурса,  да и другие файлы настроек тоже, то наверно правильно бы хранить это все в файлах ресурсов.

ZhannaD commented on src/main/java/school/faang/user_service/config/EnvHelper.java:
```diff
@@ -0,0 +1,17 @@
+import io.github.cdimascio.dotenv.DotenvEntry;
+
+public class EnvHelper {
+    public static void loadVariables() {
```
I think it's better to make this a non-static function.

lexsandr666 replied:
Соглашусь, думаю не зачем хранить этот "мусор" после старта приложения. Поправил

ZhannaD commented on src/main/java/school/faang/user_service/UserServiceApplication.java:
```diff
@@ -6,12 +6,16 @@
-    public static void main(String[] args) {
+    public static void main(String[] args)
+    {
+        EnvHelper.loadVariables();
```
better to do it in @PostConstruct public void init() { envHelper.loadVariables(); }

lexsandr666 replied:
Незнал про такую нотацию. Действительно, такое решение более локаничней и правильней. Спасибо. Поправил.

ZhannaD reviewed (commented):
Good luck :)

bortsovamaria commented:
Кажется пр не подтянулся в задачу в jira

lexsandr666 committed:
BJS2-55521 :
1) Made the loadVariables method non-static
2) Moved the loadVariables method from main to init method with @PostConstruct annotations

miked4king reviewed (commented):
Привет, в целом задача выполнена, молодец.
Со своей стороны предложу наиболее оптимальное решение, данный функционал можно сделать без добавления либы DotEnv.

у спринга есть аннотация @PropertySource("путь к .env файлу")
ее можно добавить к стартовому классу приложения, файл .env у тебя уже есть, все остальное можно выпилить

нам не придется писать лишний код EnvHelper.

lexsandr666 committed:
BJS2-55521 :
1) Removed EnvHelper class. Using @PropertySource annotation instead.
2) Removed all unnecessary methods.
3) Removed DotEnv library reference in build.gradle.kts.
